core/types/deposit_tx.go

```diff
@@ -66,7 +66,7 @@ const DepositsNonce uint64 = 0xffff_ffff_ffff_fffd
 
 // accessors for innerTx.
 func (tx *DepositTx) txType() byte           { return DepositTxType }
-func (tx *DepositTx) chainID() *big.Int      { panic("deposits are not signed and do not have a chain-ID") }
+func (tx *DepositTx) chainID() *big.Int      { return common.Big0 }
```

Comment:
Don't feel like you need to change this, but I do find it interesting that in `deriveChainId`, they allocate a new empty big int each time they grab the chain Id while they just hand out pointers in other methods.